saga/sagaState.go

```diff
@@ -478,3 +478,47 @@ func makeSagaState(sagaId string, job []byte) (*SagaState, error) {
 
 	return state, nil
 }
+
+/*
+ * Custom ToString function for SagaState
+ */
+func (state *SagaState) String() string {
+
+	fmtString := "{ SagaId: %v, " +
+		"SagaAborted: %v, " +
+		"SagaCompleted: %v, " +
+		"Tasks: [ "
+
+	for _, id := range state.GetTaskIds() {
+
+		taskState := ""
+
+		if state.IsTaskStarted(id) {
+			taskState += "Started|"
+		}
+		if state.IsTaskCompleted(id) {
+			taskState += "Completed|"
+		}
+		if state.IsCompTaskStarted(id) {
+			taskState += "CompTaskStarted|"
+		}
+		if state.IsCompTaskCompleted(id) {
+			taskState += "CompTaskCompleted|"
+		}
+
+		if len(taskState) >= 3 {
```

Comment:
prints out the state of the task, what messages we have received for it.  Below is an example of a Pretty Print.

SagaId: 26x1b7fwo85vabt9k, SagaAborted: false, SagaCompleted:
   false, Tasks: [ p2ns3hmpz717zc: Started|Completed, cnexxoah1h5ufdi:
   Started, c1y0wvdmzqw86vrul: Started|Completed, 2qi7vv9sjwgjj: Started,
   qo8q79d6l3z4x: Started|Completed, 9wfhbdeq: Started, mtle3rv2f4mgy3:
   Started, 2ttmhtwz3zn9o3p: Started, day4syhcut4hm9l1fm1g: Started,
   od0px1zafg0: Started, 43qb5ivpj0f3wpd5l5: Started|Completed,
   8p3rftaj30qvnk6h: Started, 0l5qbxzdssf3j: Started, npole3cq5j11ffxuwcg:
   Started, i1ike18d41ihdt: Started, tk6o6m: Started, ggyrpdblvbhqd1usw:
   Started|Completed, 4e9gbvn72orkwz: Started, q9nko7gzdnjjlw:
   Started|Completed, m4jlce: Started|Completed, a0ad48683x83v5: Started,
   od8opobu8e7unf: Started, aopqp53zf: Started|Completed, 8:
   Started|Completed, ntn1l047g6z: Started|Completed, ieujyl: Started,
   1k04woxx9huwc4: Started|Completed, j: Started, mah0f73rn4ya: Started,
   iaan0lmf5l04tmv: Started|Completed, 0gkxyte8mzcxcxefaz:
   Started|Completed, 46fn81v78x69190: Started|Completed, qkc6t93b6ryeu:
   Started, 5nj9m64fdvfqxzq1j: Started|Completed, wu9f4um6lfiyz08vs65k:
   Started, bi: Started|Completed, bgt8jg3pn: Started|Completed, v9b0i:
   Started, zfz: Started, 1ll: Started, ]